### PR TITLE
Prod fixes

### PR DESCRIPTION
**Pull request by CityVibes**
Adding organization endpoints back with deprecation message on use.

### CODE REVIEW

File: cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleCloud.java [hunk 1]
```diff
@@ -232,7 +232,7 @@ public void signUpAndLogInWithCustomer(SignUpInfo signUpInfo, Integer productId)
             throws ParticleCloudException {
         if (!all(signUpInfo.getUsername(), signUpInfo.getPassword(), productId)) {
             throw new IllegalArgumentException(
-                    "Email, password, and organization must all be specified");
+                    "Email, password, and product slug must all be specified");
```
Comment (idokleinman) [inline cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleCloud.java]: product Id not slug

**Review by idokleinman — CHANGES_REQUESTED** [review]
I am thinking lib should check - if `organization=true` and `productMode=false` - if so then error out and crash the app. Also if product is true and productId not set (0)... Is there a graceful way to do it in Android?

File: cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleCloud.java (hunk 1 above)
Comment (CityVibes) [reply cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleCloud.java]: True, changed that now.

**Comment by CityVibes** [comment]
Both organization=true and productMode=false would be in setup library, so I'll check if there's graceful way to crash once I update it. Initially I'd say it's possible as we're obviously accessing those fields during setup.

**Pull request merged**